source/daplink/validation.c

```diff
@@ -23,6 +23,34 @@
 #include "string.h"
 #include "target_config.h"
 
+#ifdef TARGET_MCU_CORTEX_A
+
+uint8_t validate_bin_nvic(const uint8_t *buf)
+{
+    // Very dirty hacking here for ARMv7-A (non Cortex-M) binary detection
+#ifdef BINARY_DETECTION
+    // This returns validated result when start instrunction
+    // of the buffer is BINARY_DETECTION (LDR  PC, Label)
+    // Compared with the high-order 3byte
+    if ((buf[1] == ((BINARY_DETECTION >> 8)  & 0xFF))
+     && (buf[2] == ((BINARY_DETECTION >> 16) & 0xFF))
+     && (buf[3] == ((BINARY_DETECTION >> 24) & 0xFF))) {
```

Comment:
This should probably be overridden at a board or target level with a weakref rather than with an #ifdef.